client/azure_client_set.go

```diff
@@ -63,117 +65,185 @@ func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
 
 	c := &clientConfig{
 		subscriptionID:          config.SubscriptionID,
+		partnerIdUserAgent:      fmt.Sprintf("pid-%s", config.PartnerID),
 		resourceManagerEndpoint: env.ResourceManagerEndpoint,
 		servicePrincipalToken:   servicePrincipalToken,
 	}
 
+	deploymentsClient, err := newDeploymentsClient(c)
+	if err != nil {
+		return nil, err
```

Comment:
It should be like that everywhere:

```suggestion
		return nil, microerror.Mask(err)
```

Reply:
OK, I'm slow today 😭 Fixed now.